api/novelist.py

```diff
@@ -539,31 +539,34 @@ def create_item_object(self, object, currentIdentifier, existingItem):
             isbn = object[0]
         elif object[2] is not None:
             isbn = object[2]
+        else:
+            # We cannot find an ISBN for this work -- probably due to
+            # a data error.
+            return (None, None, None, False)
 
         role = object[5]
         author = object[6] if role in Contributor.AUTHOR_ROLES else ""
 
         # If we encounter an existing ISBN and its role is "Primary Author",
         # then that value overrides the existing Author property.
+        #
+        # TODO: add 'narrator' field when we encounter a Narrator role.
         if isbn == currentIdentifier and existingItem:
             if role == Contributor.PRIMARY_AUTHOR_ROLE:
-                existingItem['Author'] = author
-                existingItem['Role'] = role
+                existingItem['author'] = author
+                existingItem['role'] = role
             return (currentIdentifier, existingItem, None, False)
         else:
             # If we encounter a new ISBN, we take whatever author value is
             # initially given to us.
             title = object[3]
             mediaType = self.medium_to_book_format_type_values.get(object[4], "")
-            narrator = object[6] if role == Contributor.NARRATOR_ROLE else ""
-
             newItem = dict(
-                ISBN=isbn,
-                Title=title,
-                MediaType=mediaType,
-                Author=author,
-                Role=role,
-                Narrator=narrator
+                isbn=isbn,
+                title=title,
+                mediaType=mediaType,
+                author=author,
+                role=role,
```

Comment:
I don't see role in the API documentation.

Reply:
It's removed from the final document, but we keep track of it so we know whether or not we've found the primary author.

I don't think we need to keep track of it _in the document_, but I didn't want to change that.